### PR TITLE
Introduces time table

### PR DESCRIPTION
**Pull request by mateuszlichon**
Adds TimeTable entity
Sets relations of TimeTable and School
Develops options for managing lessons start time and breaks length
Adds methods of exporting breaks time as a list to TimeTable entity
Adds methods of exporting time table as a list of Strings to present in views
Incorporates time tables in teacher and student view

TODO in next step:
Allow school admin to manage starting time and breaks length

### CODE REVIEW

File: src/main/java/pl/schoolmanager/entity/School.java [hunk 1]
```diff
@@ -112,5 +118,32 @@ public void setStudent(Set<Student> student) {
...
+		List<String> timeTableList = new ArrayList<>();
+		LocalTime first = tt.getStart();
+		LocalTime start = first;
+		LocalTime finish = start.plusMinutes(45);
+		timeTableList.add(start + " - " + finish);
+		
+		for (int i = 0; i < 7; i++) {
+			start = finish.plusMinutes(tt.getList().get(i));
+			finish = start.plusMinutes(45);
+			timeTableList.add(start + " - " + finish);
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/entity/School.java:144]: you can do like this: tt.getBreaks().get(i);

or iterate over map like this:
for(Map.Entry<Integer, Integer> entry : tt.getBreaks().entrySet()) {
    start = finish.plusMinutes(entry.getValue());
    // code
}

File: src/main/java/pl/schoolmanager/entity/TimeTable.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	public TimeTable() {
+		super();
+		this.start = LocalTime.of(7, 30);
+		this.breaks.put(1, 5);
+		this.breaks.put(2, 10);
+		this.breaks.put(3, 5);
+		this.breaks.put(4, 10);
+		this.breaks.put(5, 5);
+		this.breaks.put(6, 15);
+		this.breaks.put(7, 5);
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/entity/TimeTable.java:42]: i would extract it into private method

File: src/main/java/pl/schoolmanager/entity/School.java [hunk 3]
```diff
@@ -112,5 +118,32 @@ public void setStudent(Set<Student> student) {
...
+	}
+
+	public void setTimeTable(TimeTable timeTable) {
+		this.timeTable = timeTable;
+	}
+	
+	public List<String> getTimeTableList() {
+		TimeTable tt = this.timeTable;
+		if (tt == null) {
+			tt = new TimeTable();
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/entity/School.java:133]: it can be shorter:
if(timeTable == null) {
   timeTable = new TimeTable();
}

**Review by przemyslawkonik — APPROVED** [review]
:+1:

File: src/main/java/pl/schoolmanager/entity/School.java (hunk 3 above)
Comment (mateuszlichon) [reply src/main/java/pl/schoolmanager/entity/School.java:133]: Thanks. To be discussed.

File: src/main/java/pl/schoolmanager/entity/TimeTable.java (hunk 2 above)
Comment (mateuszlichon) [reply src/main/java/pl/schoolmanager/entity/TimeTable.java:42]: Good idea. I will include it.

**Pull request merged**